src/view-resources.js

```diff
@@ -67,6 +73,72 @@ interface ViewEngineHooks {
 * Represents a collection of resources used during the compilation of a view.
 */
 export class ViewResources {
+
+  static convention(target: Function) {
+    let config = target.resource;
+    let resource;
+    if (config) {
+      if (typeof config === 'string') {
+        // it's a custom element, with name is the resource variable
+        // static resource = 'my-element'
+        resource = new HtmlBehaviorResource();
+        resource.elementName = config;
```

Comment:
🐛 😱

Reply:
Rethinking about it, probably I will have to do it differently: if `name` was specified, use it and lower case it, otherwise hyphenate the class name. It's more consistent with current way how it works.

Reply:
I agree. It's strange that we modify an explictely defined name.

Hypenate made perfect sense when we derived the element from class name.